backend/collab_service/server.js

```diff
@@ -20,20 +27,71 @@ io.on('connection', (socket) => {
     console.error('Socket error:', err);
   });
 
+
   // Joining a specific collaboration session
-  socket.on('joinSession', (sessionId) => {
+  /*socket.on('joinSession', ({ sessionId, userId }) => {
     socket.join(sessionId);
     console.log(`User ${socket.id} joined session ${sessionId}`);
+
+    // Emit the sessionId and userId back to the client
+    socket.emit('sessionJoined', { sessionId, userId: socket.id });
   });
+  */
+
+  /*socket.on('matched', async (data) => {
+    console.log('Received matched data:', data);
+    // Here, emit the question data to the corresponding user in the collaboration session
+    data.sessionData.uid = socket.id; // Get the userId from sessionData
+    socket.to(userId).emit('questionDataReceived', data.questionData);
+  });
+  */
+
+  socket.on('sessionJoined', (sessionId) => {
```

Comment:
Can consider moving all these socket.io lines of code to another file, like a socketController etc, so that we don't operpopulate our main file with collab functionality.